tests/core/pyspec/eth2spec/test/gloas/block_processing/test_process_attestation.py

```diff
@@ -0,0 +1,458 @@
+    sign_attestation(spec, state, attestation)
+
+    # Verify the intended blockroot conditions
+    is_matching_blockroot = attestation.data.beacon_block_root == spec.get_block_root_at_slot(
+        state, spec.Slot(attestation.data.slot)
+    )
+    is_current_blockroot = attestation.data.beacon_block_root != spec.get_block_root_at_slot(
+        state, spec.Slot(attestation.data.slot - 1)
+    )
+    assert is_matching_blockroot is False, "Expected is_matching_blockroot = False"
+    assert is_current_blockroot is True, "Expected is_current_blockroot = True"
+
+    assert spec.is_attestation_same_slot(state, attestation.data) is False
+    yield from run_attestation_processing(spec, state, attestation, valid=True)
+
+
+@with_gloas_and_later
+@spec_state_test
+def test_valid_same_slot_attestation_index_zero(spec, state):
+    """
+    Test that attestation with index = 0 is still valid in Gloas for same slot
+    """
+    attestation_slot = 2
+    slot_2_block_root = _setup_same_slot_scenario(spec, state, target_slot=attestation_slot)
+    attestation = get_valid_attestation(
+        spec, state, slot=attestation_slot, beacon_block_root=slot_2_block_root
+    )
+    attestation.data.index = 0
+    sign_attestation(spec, state, attestation)
+
+    assert spec.is_attestation_same_slot(state, attestation.data) is True
+    yield from run_attestation_processing(spec, state, attestation, valid=True)
+
+
+@with_gloas_and_later
+@spec_state_test
+def test_invalid_same_slot_attestation_index_one(spec, state):
+    """
+    Test that same-slot condition with index = 1 is invalid (same-slot must use index = 0)
+    """
+    attestation_slot = 2
+    slot_2_block_root = _setup_same_slot_scenario(spec, state, target_slot=attestation_slot)
+    attestation = get_valid_attestation(
+        spec, state, slot=attestation_slot, beacon_block_root=slot_2_block_root
+    )
+    attestation.data.index = 1
+    sign_attestation(spec, state, attestation)
+
+    assert spec.is_attestation_same_slot(state, attestation.data) is True
+    yield from run_attestation_processing(spec, state, attestation, valid=False)
+
+
+@with_gloas_and_later
+@spec_state_test
+def test_builder_payment_weight_tracking(spec, state):
+    """
+    Test that builder payment weights are tracked correctly for Gloas
+    """
+    transition_to_slot_via_block(spec, state, 2)
+
+    # Create attestation for slot 0
+    attestation_slot = 0
+    attestation = get_valid_attestation(spec, state, slot=attestation_slot, index=0)
+    attestation.data.index = 0  # Same-slot (slot 0) must use index 0
+
+    # Get only the first validator to attest
+    committee = spec.get_beacon_committee(state, attestation_slot, 0)
+
+    # Clear all bits except first validator
+    for i in range(len(attestation.aggregation_bits)):
+        attestation.aggregation_bits[i] = i == 0
+
+    attesting_validator_index = committee[0]
+    assert state.validators[attesting_validator_index].effective_balance > 0
```

Comment:
just a nit, this is used below, so probably better to name this variable here as `expected_weight_increase` and then assert it's higher than 0.